Function rewriting in guidance decorator is broken on Python 3.14

The most direct way to see this is to run this test script:


```python
import inspect
from guidance.library._sequences import exactly_n_repeats

print(inspect.signature(exactly_n_repeats.f))
```

If you run the script on Python 3.13, it will (correctly) print:

```
(model, value, n_repeats: int)
```

But on Python 3.14.0, it prints:

```
(format, /)
```

You can also trigger the same issue by running the guidance unit tests on Python 3.14:

```
goldbaum at Nathans-MBP in ~/Documents/guidance on main
± python -m pytest tests/unit/library/test_sequences.py -sx --pdb
============================== test session starts ===============================
platform darwin -- Python 3.14.0, pytest-8.4.2, pluggy-1.6.0
rootdir: /Users/goldbaum/Documents/guidance
configfile: pyproject.toml
plugins: anyio-4.11.0, asyncio-1.3.0, cov-7.0.0
asyncio: mode=Mode.STRICT, debug=False, asyncio_default_fixture_loop_scope=None, asyncio_default_test_loop_scope=function
collected 112 items

tests/unit/library/test_sequences.py F
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> traceback >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

self = <tests.unit.library.test_sequences.TestExactlynRepeats object at 0x1290234d0>

    def test_smoke(self):
>       grammar = "AAA" + exactly_n_repeats("b", 4) + "BBB"
                          ^^^^^^^^^^^^^^^^^^^^^^^^^

tests/unit/library/test_sequences.py:11:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
guidance/_guidance.py:66: in __call__
    return self._impl(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

args = ('b', 4), kwargs = {}, reference = None
token = <Token used var=<ContextVar name='in_stateless_context' default=False at 0x104e0a070> at 0x129571f40>
no_args = False

    @functools.wraps(f)
    def wrapped(*args, **kwargs):
        # make a stateless grammar if we can
        if stateless is True or (callable(stateless) and stateless(*args, **kwargs)):
            # if we have a (deferred) reference set, then we must be in a recursive definition and so we return the reference
            reference = getattr(thread_local, "_self_call_reference_", None)
            if reference is not None:
                return reference

            # otherwise we call the function to generate the grammar
            else:
                # set the stateless context variable so that others can detect that we're currently calling a stateless function
                token = _in_stateless_context.set(True)

                # set a RuleRefNode for recursive calls (only if we don't have arguments that might make caching a bad idea)
                no_args = len(args) + len(kwargs) == 0
                if no_args:
                    thread_local._self_call_reference_ = RuleRefNode()

                try:
                    # call the function to get the grammar node
>                   node = f(_null_grammar, *args, **kwargs)
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
E                   TypeError: exactly_n_repeats() takes 1 positional argument but 3 were given

guidance/_guidance.py:160: TypeError
>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> entering PDB >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>

>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>> PDB post_mortem >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>
> /Users/goldbaum/Documents/guidance/guidance/_guidance.py(160)wrapped()
-> node = f(_null_grammar, *args, **kwargs)
(Pdb) import inspect
(Pdb) inspect.signature(f)
<Signature (format, /)>
(Pdb) f
<function exactly_n_repeats at 0x129488300>
```

Not sure what changed in Python. Maybe something in `strip_multiline_string_indents` is broken?

--- a/guidance/_utils.py
+++ b/guidance/_utils.py
@@ -131,8 +131,10 @@ def strip_multiline_string_indents(f):
 
     # find the code block
     for i in range(len(new_code_obj.co_consts)):
-        if str(type(new_code_obj.co_consts[i])) == "<class 'code'>":
+        if type(new_code_obj.co_consts[i]) == types.CodeType and new_code_obj.co_consts[i].co_name == f.__name__:
             break
+    else:
+        raise RuntimeError("Could not find function code object in modified code!")
 
     # create a new function based on the modified code
     new_f = types.FunctionType(

